truss/templates/server/common/truss_server.py

```diff
@@ -279,6 +281,22 @@ def create_application(self):
             },
         )
 
+        class RequestIdFilter(logging.Filter):
+            thread_local = threading.local()
+
+            def filter(self, record):
+                record.request_id = getattr(self.thread_local, "request_id", "UNKNOWN")
+                return True
+
+        async def intercept_request_id(request, call_next):
+            request_id = request.headers.get("x-baseten-request-id", uuid.uuid4())
+            request_id_filter = RequestIdFilter()
+            request_id_filter.thread_local.request_id = request_id
+
+            logging.getLogger("uvicorn.access").addFilter(request_id_filter)
```

Comment:
this flow is a little bit confusing to me, could you add some comments explaining why you need to _both_ do `loguru_logger.contextualize` as well as ` record.request_id = getattr(self.thread_local, "request_id", "UNKNOWN")`